src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -63,6 +63,9 @@ public function getProperties($class, array $context = array())
         foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
             $propertyName = $this->getPropertyName($reflectionMethod->name);
             if ($propertyName) {
+                if (!isset($properties[$propertyName])) {
+                    $propertyName = lcfirst($propertyName);
```

Comment:
So are we lowercasing the first letter unless it doesn't contain any lowercase letters at all? Example...

| Input | Result |
|---|---|
| `A` | `a` |
| `Aa` | `aa` |
| `AAa` | `aAa` |
| `AAA` | `AAA` |

I don't think it's a good solution, unless I misunderstood what you meant.